### ISSUE TITLE
install go1.7 ERROR

### ISSUE THREAD

**Issue by mymtwcom** [moovweb/gvm#221]
```
$ gvm install go1.7
Downloading Go source...
Installing go1.7...
 * Compiling...
ERROR: Failed to compile. Check the logs at /home/blahblahblah/.gvm/logs/go-go1.7-compile.log
ERROR: Failed to use installed version

$ cat /home/blahblahblah/.gvm/logs/go-go1.7-compile.log
##### Building Go bootstrap tool.
cmd/dist
ERROR: Cannot find /home/blahblahblah/go1.4/bin/go.
Set $GOROOT_BOOTSTRAP to a working Go tree >= Go 1.4.
```
I have go 1.6 on my ubuntu already, do I need to install go 1.4 yet in gvm, that to install go 1.7 there? wat?
![black-man-with-question-marks-wtf-meme-does-not-understand-nigga](https://cloud.githubusercontent.com/assets/2866730/17895766/c9a03a18-6956-11e6-9b03-89b7431280d3.jpg)

**Comment by cin210** [comment]
**tl;dr just set $GOROOT first**

It's a little silly. However, you don't need to downgrade go, 1.4 is just the minimum version required.

You just need to tell the go boostrap tool (separate from gvm) where a [valid Go installation (>=1.4) lives, so it can compile the latest Go you are trying to install](https://golang.org/src/cmd/dist/README)

$GOROOT_BOOSTRAP is set to $GOROOT if $GOROOT_BOOSTRAP is not defined.

If you set GOROOT_BOOTSTRAP to the path where 1.6 is installed, it should compile just fine.

ex: I have installed 1.6 at /Drive/username/.gvm/gos/go1.6/bin/go. To upgrade, do 
    
    export GOROOT_BOOSTRAP=/Users/username/.gvm/gos/go1.6
    gvm install go1.7

which is equivalent to
    
    export GOROOT=/Users/username/.gvm/gos/go1.6
    gvm install go1.7

This is because Go >1.4 compiles using Go.

**Comment by mymtwcom** [comment]
added `$GOROOT` in `.bashrc` looks like it helped, thx.
I'm closing this

**Comment by filipevarjao** [comment]
But it still getting the same error if you need recompile the entire language using all.bash

**Comment by janvdp** [comment]
I installed go (1.6) from the package manager (on Ubuntu 16.04) , and and ran:

export GOROOT_BOOTSTRAP=/usr/lib/go-1.6

after that I was able to:

gvm install go1.9.2.
